test/intl402/DateTimeFormat/constructor-options-toobject.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018 Ujjwal Sharma. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-initializedatetimeformat
+description: >
+  Tests that Intl.DateTimeFormat contructor converts the options argument
+  to an object using `ToObject` (7.1.13).
+info: |
+  12.1.2 ToDateTimeOptions
+
+  1. If options is undefined, let options be null; otherwise let options be ?
+  ToObject(options).
+---*/
+
+const toObjectResults = [
+  [true, new Boolean(true)],
+  [42, new Number(42)],
+  ['foo', new String('foo')],
+  [{}, {}],
+  [Symbol(), Object(Symbol())]
+];
+
+// Test if ToObject is used to convert primitives to Objects.
+toObjectResults.forEach(pair => {
```

Comment:
nit: you could use destructuring in the args:
```suggestion
toObjectResults.forEach(([value, result]) => {
```